src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java

```diff
@@ -103,27 +102,31 @@ public void exportProcesses(SubmissionEnvelope envelope) {
 
         ExportJob exportJob = exportJobService.createExportJob(envelope, exportJobRequest);
 
+        updateDcpVersionAndSendMessageForEachProcess(assayingProcessIds, exportJob);
+
+    }
+
+    private void updateDcpVersionAndSendMessageForEachProcess(Collection<String> assayingProcessIds, ExportJob exportJob) {
+        int totalCount = assayingProcessIds.size();
+        IndexCounter counter = new IndexCounter();
```

Comment:
This counter could also hold the `totalCount` and generate the new `ExperimentProcess`es, can't it?

Reply:
Applied with your suggestion on using a factory method.